src/main/scala/bubblewrap/PageParser.scala

```diff
@@ -11,15 +11,13 @@ import scala.collection.JavaConversions._
 import scala.util.Try
 
 case class Content(url: WebUrl, content: Array[Byte], contentType: Option[String] = None, contentCharset: Option[String] = None, contentEncoding: Option[String] = None) extends ContentType {
-  def asString: String = {
-    if (isGzip(this)) {
-      asString(Try(decompress(content)).getOrElse(content))
-    } else {
-      asString(content)
-    }
-  }
+  val asString: String = new String(asBytes, contentCharset.getOrElse("UTF-8").toUpperCase)
```

Comment:
fyi - this means, our in-mem footprint of these `Content` instances will double at a minimum. I forgot this part when you posted your query on slack. Pl ensure you've enough young-gen to handle this!

Reply:
@anant-indix and me had an offline conversation. Right now, cannonball is configured with 12G heap, out of which less than 6.5G is used today. Of this, 3.2G is in young gen. So, we clearly have enough memory to handle this excess allocation.